src/main/scala/com/comcast/xfinity/sirius/uberstore/data/UberDataFile.scala

```diff
@@ -50,7 +79,9 @@ object UberDataFile {
    *
    * @param dataFileName the name of the file for this descriptor to wrap
    */
-  private[data] class UberFileDesc(dataFileName: String) {
+  private[data] class UberFileDesc(dataFileName: String,
+                                   useReadBuffer: Boolean,
+                                   readBufferSize: Int) {
```

Comment:
I think the `UberFileDesc` is actually the place to differentiate between the types of `UberDataFileReader` here. The `createReadHandle` can use the buffer type and choose whether to make a `BufferedUberDataFileReader` or a `RandomAccessUberDataFileReader`. Holding the config values just to pass them into an `apply` later doesn't seem to make lots of sense, and this class is all about generating these things anyway.